File: simple_distance_algorithm.py
import math
import time





def vel_function(vx,vy):
  velocity = math.sqrt(vx*vx + vy*vy)
  return velocity

# A negative distance means the drone is moving away from the object
def distance_function(obj_width1, obj_width2, time_lapse,velocity):
  if type(obj_width1 == int):
      obj_width1 = float(obj_width1)
      
  if type(obj_width2 == int):
      obj_width2 = float(obj_width2)
      
  if type(velocity == int):
      velocity = float(velocity)

  
  if type(time_lapse == int):
      time_lapse = float(time_lapse)
      
  dis_trav = velocity*time_lapse
  
  if (obj_width1 > obj_width2):
      d = dis_trav + dis_trav/(obj_width1/obj_width2 -1) 
      obj_dist = d
  elif (obj_width1 < obj_width2):
      d = dis_trav/(1.0 - obj_width1/obj_width2)
      obj_dist = d-dis_trav
  else:
      print("no distance estimation can be done at this moment")
      obj_dist = float("nan")
       
  return obj_dist

t0 = time.time()
print(distance_function(5,8,0.1,5))
#print(vel_function(5,5))
t1 = time.time()

total = t1-t0

print (total)

File: Simple_distance_algorithm.c
/*
 * Simple_distance_algorithm.c
 *
 *  Created on: Mar 18, 2021
 *      Author: vi-hung
 */

#include <stdio.h>
#include <math.h>

/*function declaration*/
double vel_function(double, double);
double distance_function(double,double,double,double);



int main(int argc, char *argv[]) {
	//double vel = vel_function(5.0,5.0);
		double dist = distance_function(8,8,0.1,5);

		printf("Distance to object is %lf\n",dist);
	}


double vel_function(double vx, double vy){
	double d = vx*vx + vy*vy;
	return sqrt(d);
}

double distance_function(double obj_width1, double obj_width2, double time_lapse, double velocity){

  double dis_trav = velocity*time_lapse;
  double d;
  double obj_dist;

  if (obj_width1 > obj_width2){
     d = dis_trav + dis_trav/(obj_width1/obj_width2 -1);
     obj_dist = d;
  }
  else if (obj_width1 < obj_width2){
      d = dis_trav/(1.0 - obj_width1/obj_width2);
      obj_dist = d-dis_trav;
  }
  else{
      obj_dist = NAN;
  }

  return obj_dist;
}


